sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java

```diff
@@ -0,0 +1,234 @@
+    private static final String topicName = "my-topic-name" + randomPadding();
+    private static final String eventSubscriptionName = "event-subscription" + randomPadding();
+    private static final String eventHubRuleName = "my-management-rule" + randomPadding();
+
+    /**
+     * Main entry point.
+     *
+     * @param args the parameters
+     */
+    public static void main(String[] args) {
+        try {
+            // 1. Authenticate
+            // Authentication environment variables need to be set: AZURE_CLIENT_ID,AZURE_TENANT_ID,AZURE_CLIENT_SECRET,AZURE_SUBSCRIPTION_ID
+            TokenCredential credential = new EnvironmentCredentialBuilder()
+                .authorityHost(AzureAuthorityHosts.AZURE_PUBLIC_CLOUD)
+                .build();
+
+            AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+            // 2. Create ResourceManager, EventHubsManager, EventGridManager
+            resourceManager = AzureResourceManager.configure()
+                .withLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .authenticate(credential, profile)
+                .withDefaultSubscription();
+
+            eventHubsManager = EventHubsManager.configure()
+                .withLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS)
+                .authenticate(credential, profile);
+
+            eventGridManager = EventGridManager.configure()
+                .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BODY_AND_HEADERS))
+                .authenticate(credential, profile);
+
+            // 3. Run sample
+            runSample();
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+            e.printStackTrace();
+        }
+    }
+
+    private static void runSample() {
+        try {
+
+            // 1. Create a resource group.
+            ResourceGroup resourceGroup =
+                resourceManager.resourceGroups().define(resourceGroupName).withRegion(Region.US_EAST).create();
+
+            System.out.println("Resource group created with name " + resourceGroupName);
+
+            // 2. Create an event hub.
+            // 2.1 Create a event hub namespace.
+            EventHubNamespace namespace = eventHubsManager.namespaces()
+                .define(eventHubNamespace)
+                .withRegion(region)
+                .withExistingResourceGroup(resourceGroupName)
+                .withAutoScaling()
+                .withSku(EventHubNamespaceSkuType.STANDARD)
+                .create();
+
+            System.out.println("EventHub namespace created with name " + namespace.name());
+
+            // 2.2 Create event hub.
+            EventHub eventHub = resourceManager.eventHubs()
+                .define(eventHubName)
+                .withExistingNamespace(resourceGroupName, eventHubNamespace)
+                .withNewManageRule(eventHubRuleName)
+                .withPartitionCount(1)
```

Comment:
Maybe you can add some comment about partition size.

Reply:
updated